assets/src/dashboard/app/views/myStories/index.js

```diff
@@ -104,22 +112,38 @@ function MyStories() {
       searchTerm: typeaheadValue,
       sortDirection: viewStyle === VIEW_STYLE.LIST && currentListSortDirection,
       status,
+      page: currentPageToFetch,
     });
   }, [
     viewStyle,
     currentListSortDirection,
+    currentPageToFetch,
     currentStorySort,
     fetchStories,
     status,
     typeaheadValue,
   ]);
 
-  const filteredStories = useMemo(() => {
-    return stories.filter((story) => {
-      const lowerTypeaheadValue = typeaheadValue.toLowerCase();
-      return story.title.toLowerCase().includes(lowerTypeaheadValue);
+  const orderedStories = useMemo(() => {
+    return storiesOrderById.map((storyId) => {
+      return stories[storyId];
     });
-  }, [stories, typeaheadValue]);
+  }, [stories, storiesOrderById]);
+
+  // control pagination and attempts to load further pages
+  // also controls the loadings vs done messaging
+  // could be in reducer
+  useEffect(() => {
+    setAllPagesLoaded(currentPageToFetch >= totalPages);
+  }, [currentPageToFetch, totalPages]);
```

Comment:
this seems like `allPagesLoaded` is derived state. which make make sense for it to be a simple value:
```
const allPagesLoaded = currentPageToFetch >= totalPages;
```
since it's a primitive we shouldn't need to memoize this because it will work with shallow comparison. also adding a memo around this would probably less performant than the simple boolean operation it's computing on each render (useMemo's using the same boolean operation under the hood to tell if it should update the memoized value)

Reply:
I am kind of playing with the idea that we have an `allPagesFetched` value in our stories reducer that tracks based on the page just fetched and the total pages, since this is something at the fetched stories level not the view and then using that value via context state and just cutting out this whole set up on the view level.  

I made this change in my last push up if you want to look.

Reply:
awesome. yea that works too! will check out